### PR TITLE
410 backend UI api wrapper GET & POST policies endpoint with 02x

### PR DESCRIPTION
**Pull request by SaadEGI**
Related to  Backend UI API Wrapper: Get Policies Endpoint with 0.2.x #410

### CODE REVIEW

File: extensions/wrapper/wrapper/src/main/java/de/sovity/edc/ext/wrapper/api/ui/UiResource.java [hunk 1]
```diff
@@ -78,4 +78,19 @@ public TransferHistoryPage transferHistoryPageEndpoint() {
...
+
+    @GET
+    @Path("pages/policy-page")
+    @Produces(MediaType.APPLICATION_JSON)
+    public PolicyPage policyPage() {
+        return new PolicyPage(policyApiService.getPolicies());
+    }
+
+    @POST
+    @Path("pages/policy-page")
```
Comment (richardtreier) [inline extensions/wrapper/wrapper/src/main/java/de/sovity/edc/ext/wrapper/api/ui/UiResource.java:90]: Nit: Maybe POST pages/policy/page/policies would make more sense, as policies would be the resource, and "pages/policy-page" is just one entry on the resource "pages"

File: extensions/wrapper/wrapper/src/main/java/de/sovity/edc/ext/wrapper/api/ui/UiResource.java [hunk 2]
```diff
@@ -78,4 +78,19 @@ public TransferHistoryPage transferHistoryPageEndpoint() {
...
+    @Produces(MediaType.APPLICATION_JSON)
+    public IdResponse createPolicy(UiPolicyDto policyDto) {
```
Comment (richardtreier) [inline extensions/wrapper/wrapper/src/main/java/de/sovity/edc/ext/wrapper/api/ui/UiResource.java:92]: For the createPolicy call we need a UiPolicyCreateDto, which does not contain the legacy policy nor the errors. Meaning we are cutting down the API Model for the create process to what it exactly needs to the behavior will be exactly defined.

The response for this call could actually return a UiPolicyDto, which would make testing easier

File: extensions/wrapper/wrapper/src/main/java/de/sovity/edc/ext/wrapper/api/ui/pages/policy/PolicyApiService.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+import org.eclipse.edc.policy.model.Policy;
+import org.eclipse.edc.spi.query.QuerySpec;
+
+import java.util.List;
+
+
+@RequiredArgsConstructor
+public class PolicyApiService {
+
+
```
Comment (richardtreier) [inline extensions/wrapper/wrapper/src/main/java/de/sovity/edc/ext/wrapper/api/ui/pages/policy/PolicyApiService.java:34]: newlines

File: extensions/wrapper/wrapper/src/main/java/de/sovity/edc/ext/wrapper/api/ui/pages/policy/PolicyApiService.java [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    private final PolicyDefinitionService policyDefinitionService;
+    private final PolicyMapper policyMapper;
+
+    public List<UiPolicyDto> getPolicies() {
+        var policies = getAllPolicies();
+
+        return policies.stream().map(policy -> {
```
Comment (richardtreier) [inline extensions/wrapper/wrapper/src/main/java/de/sovity/edc/ext/wrapper/api/ui/pages/policy/PolicyApiService.java:41]: we should have a defined sorting

File: extensions/wrapper/wrapper/src/main/java/de/sovity/edc/ext/wrapper/api/ui/pages/policy/PolicyApiService.java [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+            var entry = policyMapper.buildPolicyDto(policy);
+            return entry;
+        }).toList();
+    }
+
+    private List<Policy> getAllPolicies() {
```
Comment (richardtreier) [inline extensions/wrapper/wrapper/src/main/java/de/sovity/edc/ext/wrapper/api/ui/pages/policy/PolicyApiService.java:47]: With all the business/product/page naming vs core edc entity naming it is understandable to get confused.

The "policy page" is for managing "policy definitions", meaning, a Policy + Policy Definition ID.

You might have to rework some objects and stuff, as creating a "policy" means creating a "policy definition", which will also require a policy definition ID.

Also the resulting objects will be PolicyDefinitions.

Whether you'll decide to rename your API objects to PolicyDefinitions or not is up to you.

E.g. when fetching another connector's catalog, you get data offers, which contain a policy. At that point you won't have the other connector's internal policy definition's ID. However, when you create a contract definition, the only way you can supply your policy is by specifying a policy definition's ID.

Meaning:
- Policy = Policy JSON-LD, ephermal
- Policy Definition = Persisted Policy + Chosen Policy Definition ID
- Creating a Contract Definition requires a persisted Policy Definition
- Contract Negotiations contain Policies (embedded JSON only, no policy definition ID, as the data offer might come from another connector)
- Contract Offers contain Policies (embedded JSON only, no policy definition ID, the data offers definitely come from other connectors)

**Review by richardtreier — CHANGES_REQUESTED** [review]

**Comment by SaadEGI** [comment]
duplicate of https://github.com/sovity/edc-extensions/pull/481

**Pull request closed without merging**